app/models/requests/CreateClaimRequest.scala

```diff
@@ -146,9 +153,17 @@ object CreateClaimRequest {
       case _ => userAnswers.get(ImporterManualAddressPage)
     }
 
-    def getImporterName(userAnswers: UserAnswers): Option[UserName] = userAnswers.get(ClaimantTypePage) match {
-      case Some(ClaimantType.Importer) => userAnswers.get(ImporterNamePage)
-      case _ => userAnswers.get(AgentNameImporterPage)
+    def getAgentImporterName(userAnswers: UserAnswers): Option[String] = userAnswers.get(RepresentativeAgentNamePage).map(_.toString)
+
+    def getImporterName(userAnswers: UserAnswers): Option[String] = {
+      val declarantNamePage = userAnswers.get(DeclarantNamePage)
+      userAnswers.get(ClaimantTypePage) match {
+        case Some(ClaimantType.Importer) => userAnswers.get(DoYouOwnTheGoodsPage) match {
+          case Some(DoYouOwnTheGoods.No) => userAnswers.get(ImporterNamePage).map(_.value)
+          case _ => declarantNamePage.map(_.toString)
```

Comment:
same here replace case otherwise with case Some(DoYouOwnTheGoods.Yes), if not no only yes.

Reply:
again the same .. compiler warning error